src/snowflake/cli/_plugins/spcs/services/manager.py

```diff
@@ -133,12 +136,65 @@ def status(self, service_name: str) -> SnowflakeCursor:
         return self.execute_query(f"CALL SYSTEM$GET_SERVICE_STATUS('{service_name}')")
 
     def logs(
-        self, service_name: str, instance_id: str, container_name: str, num_lines: int
+        self,
+        service_name: str,
+        instance_id: str,
+        container_name: str,
+        num_lines: int,
+        since_timestamp: str = "",
+        include_timestamps: bool = False,
     ):
-        return self.execute_query(
-            f"call SYSTEM$GET_SERVICE_LOGS('{service_name}', '{instance_id}', '{container_name}', {num_lines});"
+        cursor = self.execute_query(
+            f"call SYSTEM$GET_SERVICE_LOGS('{service_name}', '{instance_id}', '{container_name}', "
+            f"{num_lines}, False, '{since_timestamp}', {include_timestamps});"
         )
 
+        for log in cursor.fetchall():
+            yield log[0] if isinstance(log, tuple) else log
+
+    def stream_logs(
+        self,
+        service_name: str,
+        instance_id: str,
+        container_name: str,
+        num_lines: int,
+        since_timestamp: str,
+        include_timestamps: bool,
+        interval_seconds: int,
+    ):
+        try:
+            prev_timestamp = since_timestamp
+            prev_log_records: List[str] = []
+
+            while True:
+                new_log_records = [
+                    log
+                    for log in self.logs(
+                        service_name=service_name,
+                        instance_id=instance_id,
+                        container_name=container_name,
+                        num_lines=num_lines,
+                        since_timestamp=prev_timestamp,
+                        include_timestamps=True,
+                    )
+                ]
+
+                if new_log_records:
+                    dedup_log_records = new_logs_only(prev_log_records, new_log_records)
+                    formatted_logs = "\n".join(
+                        filter_log_timestamp(log, include_timestamps)
+                        for log in dedup_log_records
+                    )
+                    yield formatted_logs
```

Comment:
the CLI will already print each message on it's own line, so you don't need to join them:
```suggestion
                    for log in dedup_log_records:
                        yield filter_log_timestamp(log, include_timestamps)
```